docs/guide/request_lifecycle.md

```diff
@@ -0,0 +1,252 @@
+# Request lifecycle: from request to response
+
+
+## Beginning of HTTP request
+
+When entering the architecture, the HTTP request can be handled by several component such as a firewall, a load balancer, or a reverse proxy before arriving on the web server itself.
+
+For an overview of what happens on a reverse proxy like Varnish or Fastly, see [Context-aware HTTP cache / Request lifecycle](cache/context_aware_cache/#request-lifecycle).
+
+When arriving at a web server, the request is filtered by Apache Virtual Host, Nginx Server Blocks or equivalent. This mainly (and shortly) separates requests of static resources from requests to the application.
+
+As [[= product_name =]] is a Symfony application, the handling of requests starts like in Symfony (see [Symfony and HTTP Fundamentals](https://symfony.com/doc/current/introduction/http_fundamentals.html)).
+
+If the HTTP request is to be treated by [[= product_name =]], it goes to the `public/index.php` of the [Symfony Front Controller](https://symfony.com/doc/current/configuration/front_controllers_and_kernel.html#the-front-controller).
+
+The front controller transforms the HTTP request into a PHP [`Request` object](https://symfony.com/doc/current/introduction/http_fundamentals.html#symfony-request-object) and passes it to Symfony's Kernel to get a [`Response` object](https://symfony.com/doc/current/introduction/http_fundamentals.html#symfony-response-object) that is transformed and sent back as an HTTP response.
+
+The schemas start with a regular `Request` object from a browser that enters Symfony and [[= product_name =]]. There is no ESI, no REST, and no GraphQL request performed.
+
+
+## Lifecycle flowcharts
+
+### Concept flowchart
+
+The chart below introduces the logic of the request treatment.
+
+![Simplified request lifecycle flowchart](img/request_lifecycle_concept.png)
+
+
+### Kernel events flowchart
+
+The following chart shows events, listeners and attributes added to the request or its wrapping event object.
+
+![Detailed request lifecycle flowchart organised around kernel events](img/request_lifecycle_events.png){: width="741" height="2061" style="max-height: none;"}
+
+This schema is described below event by event.
+
+!!! tip
+
+    To list all listeners that listen to an event, run `php bin/console debug:event-dispatcher <event.name>`, for example:
+    
+    ```bash
+    bin/console debug:event-dispatcher kernel.request
```

Comment:
```suggestion
    php bin/console debug:event-dispatcher kernel.request
```

Reply:
Agreed, that's the convention, I missed it.